hooks/mari/publish_mipmaps.py

```diff
@@ -0,0 +1,221 @@
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        geo_name = item.properties.mari_geo_name
+        geo = mari.geo.find(geo_name)
+        if not geo:
+            error_msg = "Failed to find geometry in the project! Validation failed." % geo_name
+            self.logger.error(error_msg)
+            return False
+
+        channel_name = item.properties.mari_channel_name
+        channel = geo.findChannel(channel_name)
+        if not channel:
+            error_msg = "Failed to find channel on geometry! Validation failed." % channel_name
+            self.logger.error(error_msg)
+            return False
+
+        layer_name = item.properties.get("mari_layer_name")
+        if layer_name:
+            layer = channel.findLayer(layer_name)
+            if not layer:
+                error_msg = "Failed to find layer for channel: %s Validation failed." % layer_name
+                self.logger.error(error_msg)
+                return False
+
+        return super(MariPublishMipmapsPlugin, self).validate(task_settings, item)
+
+
+    def publish_files(self, task_settings, item, publish_path):
+        """
+        Overrides the inherited method to export out session items to the publish_path location.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :param publish_path: The output path to publish files to
+        """
+        publisher = self.parent
+
+        # get the path in a normalized state. no trailing separator, separators
+        # are appropriate for current os, no double separators, etc.
+        target_path = sgtk.util.ShotgunPath.normalize(publish_path)
+
+        try:
+            # ensure the publish folder exists:
+            publish_folder = os.path.dirname(target_path)
+            ensure_folder_exists(publish_folder)
+
+            # get all associated files if this is a file sequence
+            source_path = item.parent.properties.publish_path
+            seq_path = publisher.util.get_frame_sequence_path(source_path)
+            if seq_path:
+                src_files = publisher.util.get_sequence_path_files(source_path)
+            else:
+                src_files = [source_path]
+
+            # write mipmaps for each file in the sequence
+            self.create_mipmaps_for_seq(src_files, target_path)
+
+        except Exception as e:
+            raise TankError("Failed to publish file for item '%s': %s" % (item.name, str(e)))
+
+        self.logger.debug(
+            "Published file for item '%s' to '%s'." % (item.name, target_path)
+        )
+
+        return [target_path]
```

Comment:
You should return the sequence of created files, not the template path.

Reply:
This is the sequence path (resolved template path). As in a path containing the seqkey (in this case UDIM and not a list of files. Eg.: `/dd/shows/HIRANYA/SHARED/TEXTURE/texture/Character/sw_test_hiranya/mipmap/v004/diffuseColor1/Base/texture_Character-sw_test_hiranya-mipmap.v004.<UDIM>.tx`

That is the pattern followed in all images (eg. the original tiff textures contain the UDIM key and renders contain %04d) We can't have a list as there is only one path associated with a published file entity.

Reply:
If you look at the return type of the super class, it is returning the expanded list of files, not the sequence path. The return value is not used for populating the `PublishedFile` path field anyways...at the moment its not used for anything, but we should maintain consistent behavior from any subclasses.